### PR TITLE
Remove Hub from GUI

### PR DESCRIPTION
**Pull request by retiutut**
Bye bye Hub..... Hello BrainFlow!

### CODE REVIEW

File: OpenBCI_GUI/Extras.pde [hunk 1]
```diff
@@ -14,6 +14,30 @@
 //                       Global Functions
 //------------------------------------------------------------------------
 
+/**
+  * @description Helper function to determine if the system is linux or not.
+  * @return {boolean} true if os is linux, false otherwise.
+  */
```
Comment (Andrey1994) [inline OpenBCI_GUI/Extras.pde:20]: I think these functions were used only by Hub and more likely we dont need them

File: OpenBCI_GUI/DataProcessing.pde [hunk 2]
```diff
@@ -17,6 +17,10 @@ final int GAMMA = 4; // 30-55 Hz
 
 float playback_speed_fac = 1.0f;  //make 1.0 for real-time.  larger for faster playback
 
+int prevBytes = 0;
```
Comment (Andrey1994) [inline OpenBCI_GUI/DataProcessing.pde:20]: I've tried to grep and looks like we can remove these variables... but maybe I am mistaken

Comment (retiutut) [reply OpenBCI_GUI/DataProcessing.pde:20]: They are used in processNewData() here in DataProcessing, so they got moved closer to the code. Adding a comment saying this.

File: OpenBCI_GUI/Extras.pde (hunk 1 above)
Comment (retiutut) [reply OpenBCI_GUI/Extras.pde:20]: Used in Control Panel getBLED112Port(). Also, sometimes things display in a different position on Linux an there is one time I do something like `int osPadding = ...`

File: OpenBCI_GUI/W_Networking.pde [hunk 3]
```diff
@@ -1785,9 +1785,8 @@ class Stream extends Thread {
                 for (int i = 0; i < NUM_ANALOG_READS; i++) {
                     msg.clearArguments();
                     msg.add(i+1);
-                    //ADD Accelerometer data
-                    msg.add(hub.validAccelValues[i]);
-                    // println(i + " | " + hub.validAccelValues[i]);
+                    //todo[brainflow] get analog data
+                    //msg.add(BRAINFLOW_ANALOG_VALS);
```
Comment (daniellasry) [inline OpenBCI_GUI/W_Networking.pde:1789]: Can you make this commented-out line say `msg.add(hub.validAccelValues[i]);`, just so we have documentation on what the old function call was

File: OpenBCI_GUI/W_Networking.pde [hunk 4]
```diff
@@ -1798,7 +1797,7 @@ class Stream extends Thread {
             } else if (this.protocol.equals("UDP")) {
                 String outputter = "{\"type\":\"auxiliary\",\"data\":[";
                 for (int i = 0; i < NUM_ANALOG_READS; i++) {
-                    int auxData = hub.validAccelValues[i];
+                    int auxData = 0; //todo[brainflow] get analog data
```
Comment (daniellasry) [inline OpenBCI_GUI/W_Networking.pde:1800]: could you keep the old line commented out just for documentation when we come back and fix this widget? same for the lines below

Comment (daniellasry) [reply OpenBCI_GUI/W_Networking.pde:1800]: Nevemind. Don't bother doing this (i walready have a networking widget PR)

File: OpenBCI_GUI/W_Networking.pde (hunk 3 above)
Comment (daniellasry) [reply OpenBCI_GUI/W_Networking.pde:1789]: Nevermind, don't bnother.

File: OpenBCI_GUI/OpenBCI_GUI.pde [hunk 5]
```diff
@@ -782,17 +620,11 @@ void initSystem() {
...
-            abandonInit = true;
-        } else {
-            //println("FOUND FIRMWARE FROM HUB == " + hub.getFirmwareVersion());
-        }
-    }
+    // todo - Check Cyton Firmware version here, if possible
+    //if (eegDataSource == DATASOURCE_CYTON) {
+        //whenfirmware is very out of date, and commands may not work, so abandon init
+        //abandonInit = true;
+    //}
```
[the remark this answers is not in the document]
Comment (retiutut) [reply OpenBCI_GUI/OpenBCI_GUI.pde:627]: Going to write this down and add to ongoing troubleshooting folder

File: OpenBCI_GUI/OpenBCI_GUI.pde [hunk 6]
```diff
@@ -499,158 +486,12 @@ private void prepareExitHandler () {
...
     ));
 }
 
-/**
-  * Starts the hub and sets prepares the exit handler.
-  */
-void hubInit() {
-    isHubInitialized = true;
-    hubStart();
-    prepareExitHandler();
```
[the remark this answers is not in the document]
Comment (retiutut) [reply OpenBCI_GUI/OpenBCI_GUI.pde:524]: done

**Comment by retiutut** [comment]
There was a conflict in prepareExitHandler().So I merged upstream, removed Hub code, and keeping 

```
// finalize any playback files
            dataLogger.onShutDown();
```

File: OpenBCI_GUI/OpenBCI_GUI.pde [hunk 7]
```diff
@@ -486,6 +488,9 @@ private void prepareExitHandler () {
                 w_networking.stopNetwork();
                 println("openBCI_GUI: shutDown: Network streams stopped");
             }
+
+            // finalize any playback files
+            dataLogger.onShutDown();
```
Comment (daniellasry) [inline OpenBCI_GUI/OpenBCI_GUI.pde]: Why did you have to add this? copyGUISampleData() is run on startup so there should not be any playback files to finalize.

Comment (retiutut) [reply OpenBCI_GUI/OpenBCI_GUI.pde]: I thought this was from a recent change. Remove it?

Comment (daniellasry) [reply OpenBCI_GUI/OpenBCI_GUI.pde]: It's supposed to be called inside the exit handler. Maybe you merged it to the wrong place?

Comment (daniellasry) [reply OpenBCI_GUI/OpenBCI_GUI.pde]: Hang on! I might have misread the code 😛

Comment (daniellasry) [reply OpenBCI_GUI/OpenBCI_GUI.pde]: Yep, you're fine!

**Comment by retiutut** [comment]
Gonna have a cookie and hit that merge button!

**Pull request merged**